src/test/java/com/example/universe/simulator/entityservice/unit/controllers/PlanetControllerTest.java

```diff
@@ -0,0 +1,311 @@
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(planetDto));
+        MockHttpServletResponse response = mockMvc
+                .perform(requestBuilder)
+                .andReturn()
+                .getResponse();
+        // assert
+        verifySuccessfulResponse(response, planetDto);
+        then(service).should().add(planet);
+    }
+
+    @Test
+    void testAdd_validateNullName() throws Exception {
+        // arrange
+        PlanetDto planetDto = TestUtils.buildPlanetDtoForAdd();
+        planetDto.setName(null);
+        // act
+        RequestBuilder requestBuilder = MockMvcRequestBuilders
+                .post("/planet/add")
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(planetDto));
+        MockHttpServletResponse response = mockMvc
+                .perform(requestBuilder)
+                .andReturn()
+                .getResponse();
+        // assert
+        verifyErrorResponse(response.getContentAsString(), ErrorCodeType.MISSING_PARAMETER_NAME);
+        then(service).should(never()).add(any());
+    }
+
+    @Test
+    void testAdd_validateEmptyName() throws Exception {
+        // arrange
+        PlanetDto planetDto = TestUtils.buildPlanetDtoForAdd();
+        planetDto.setName("");
+        // act
+        RequestBuilder requestBuilder = MockMvcRequestBuilders
+                .post("/planet/add")
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(planetDto));
+        MockHttpServletResponse response = mockMvc
+                .perform(requestBuilder)
+                .andReturn()
+                .getResponse();
+        // assert
+        verifyErrorResponse(response.getContentAsString(), ErrorCodeType.MISSING_PARAMETER_NAME);
+        then(service).should(never()).add(any());
+    }
+
+    @Test
+    void testAdd_validateBlankName() throws Exception {
+        // arrange
+        PlanetDto planetDto = TestUtils.buildPlanetDtoForAdd();
+        planetDto.setName(" ");
+        // act
+        RequestBuilder requestBuilder = MockMvcRequestBuilders
+                .post("/planet/add")
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(planetDto));
+        MockHttpServletResponse response = mockMvc
+                .perform(requestBuilder)
+                .andReturn()
+                .getResponse();
+        // assert
+        verifyErrorResponse(response.getContentAsString(), ErrorCodeType.MISSING_PARAMETER_NAME);
+        then(service).should(never()).add(any());
+    }
+
+    @Test
+    void testUpdate() throws Exception {
```

Comment:
This test doesn't check dirty input fixes. Also you're missing a test case when id is not found.